CIScripts/Tests/VRouterAgentTests.ps1

```diff
@@ -501,15 +502,67 @@ function Test-VRouterAgentIntegration {
         })
     }
 
-    function Test-MultiComputeNodesPing {
+    function Test-ICMPoMPLSoGRE {
         Param ([Parameter(Mandatory = $true)] [PSSessionT] $Session1,
                [Parameter(Mandatory = $true)] [PSSessionT] $Session2,
                [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
 
         $Job.StepQuiet($MyInvocation.MyCommand.Name, {
-            Write-Host "===> Running: Test-MultiComputeNodesPing"
+            Write-Host "===> Running: Test-ICMPoMPLSoGRE"
             Test-Ping -Session1 $Session1 -Session2 $Session2 -TestConfiguration $TestConfiguration -Container1Name "container1" -Container2Name "container2"
-            Write-Host "===> PASSED: Test-MultiComputeNodesPing"
+            Write-Host "===> PASSED: Test-ICMPoMPLSoGRE"
+        })
+    }
+
+    function Test-ICMPoMPLSoUDP {
+        Param ([Parameter(Mandatory = $true)] [PSSessionT] $Session1,
+               [Parameter(Mandatory = $true)] [PSSessionT] $Session2,
+               [Parameter(Mandatory = $true)] [TestConfiguration] $TestConfiguration)
+
+        $Job.StepQuiet($MyInvocation.MyCommand.Name, {
+            Write-Host "===> Running: Test-ICMPoMPLSoUDP"
+
+            $TestConfigurationTemp = $TestConfiguration.ShallowCopy()
+            $TestConfigurationTemp.DockerDriverConfiguration = $TestConfiguration.DockerDriverConfiguration.ShallowCopy()
+            $TestConfigurationTemp.ControllerIP = $Env:CONTROLLER_IP_UDP
+            $TestConfigurationTemp.DockerDriverConfiguration.AuthUrl = $Env:DOCKER_DRIVER_AUTH_URL_UDP
+
+            $ContrailUrl = $TestConfigurationTemp.ControllerIP + ":" + $TestConfigurationTemp.ControllerRestPort
+            $ContrailCredentials = $TestConfigurationTemp.DockerDriverConfiguration
+            $AuthToken = Get-AccessTokenFromKeystone -AuthUrl $ContrailCredentials.AuthUrl -TenantName $ContrailCredentials.TenantConfiguration.Name `
+                -Username $ContrailCredentials.Username -Password $ContrailCredentials.Password
+            $RouterIp1 = Invoke-Command -Session $Session1 -ScriptBlock {
+                return $((Get-NetIPAddress -InterfaceAlias $Using:TestConfigurationTemp.VHostName -AddressFamily IPv4).IpAddress)
+            }
+            $RouterIp2 = Invoke-Command -Session $Session2 -ScriptBlock {
+                return $((Get-NetIPAddress -InterfaceAlias $Using:TestConfigurationTemp.VHostName -AddressFamily IPv4).IpAddress)
+            }
+            $RouterUuid1 = Add-ContrailVirtualRouter -ContrailUrl $ContrailUrl -AuthToken $AuthToken -RouterName $Session1.ComputerName -RouterIp RouterIp1
+            $RouterUuid2 = Add-ContrailVirtualRouter -ContrailUrl $ContrailUrl -AuthToken $AuthToken -RouterName $Session2.ComputerName -RouterIp RouterIp2
+
+            Try {
+                Test-Ping -Session1 $Session1 -Session2 $Session2 -TestConfiguration $TestConfigurationTemp -Container1Name "container1" -Container2Name "container2"
```

Comment:
Don't I need a `catch` here? Like badly need?

Reply:
Are you thinking about catching exception and re-throwing it ? Indeed if exception is thrown in Finally statement it will suppress original one. However as long as errors in try block don't impact on Finally block, I believe it isn't that crucial. What do you think about that ?